compilation errors in v12.2 . File trafficgraphdatatests.cpp. Probably due to qt older version.

```
src@d:~/compiles/dash$ make
Making all in src
make[1]: Entering directory '/home/src/compiles/dash/src'
make[2]: Entering directory '/home/src/compiles/dash/src'
  CXX      crypto/libdashconsensus_la-hmac_sha512.lo
  CXX      crypto/libdashconsensus_la-ripemd160.lo
  CXX      crypto/libdashconsensus_la-sha1.lo
  CXX      crypto/libdashconsensus_la-sha256.lo
  CXX      crypto/libdashconsensus_la-sha512.lo
  CXX      libdashconsensus_la-hash.lo
  CXX      primitives/libdashconsensus_la-transaction.lo
  CXX      libdashconsensus_la-pubkey.lo
  CXX      script/libdashconsensus_la-dashconsensus.lo
  CXX      script/libdashconsensus_la-interpreter.lo
  CXX      script/libdashconsensus_la-script.lo
  CXX      libdashconsensus_la-uint256.lo
  CXX      libdashconsensus_la-utilstrencodings.lo
make[3]: Entering directory '/home/src/compiles/dash/src/secp256k1'
gcc -I. -g -O2 -Wall -Wextra -Wno-unused-function -c src/gen_context.c -o gen_context.o
gcc gen_context.o -o gen_context
./gen_context
  CC       src/libsecp256k1_la-secp256k1.lo
  CCLD     libsecp256k1.la
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
make[3]: Leaving directory '/home/src/compiles/dash/src/secp256k1'
  CXXLD    libdashconsensus.la
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      dashd-dashd.o
  CXX      libbitcoin_server_a-addrman.o
  CXX      libbitcoin_server_a-addrdb.o
  CXX      libbitcoin_server_a-alert.o
  CXX      libbitcoin_server_a-bloom.o
  CXX      libbitcoin_server_a-chain.o
  CXX      libbitcoin_server_a-checkpoints.o
  CXX      libbitcoin_server_a-httprpc.o
  CXX      libbitcoin_server_a-httpserver.o
  CXX      libbitcoin_server_a-init.o
  CXX      libbitcoin_server_a-dbwrapper.o
  CXX      libbitcoin_server_a-governance.o
  CXX      libbitcoin_server_a-governance-classes.o
  CXX      libbitcoin_server_a-governance-object.o
  CXX      libbitcoin_server_a-governance-validators.o
  CXX      libbitcoin_server_a-governance-vote.o
  CXX      libbitcoin_server_a-governance-votedb.o
  CXX      libbitcoin_server_a-merkleblock.o
  CXX      libbitcoin_server_a-messagesigner.o
  CXX      libbitcoin_server_a-miner.o
  CXX      libbitcoin_server_a-net.o
  CXX      libbitcoin_server_a-netfulfilledman.o
  CXX      libbitcoin_server_a-net_processing.o
  CXX      libbitcoin_server_a-noui.o
  CXX      policy/libbitcoin_server_a-fees.o
  CXX      policy/libbitcoin_server_a-policy.o
  CXX      libbitcoin_server_a-pow.o
  CXX      libbitcoin_server_a-privatesend.o
  CXX      libbitcoin_server_a-privatesend-server.o
  CXX      libbitcoin_server_a-rest.o
  CXX      rpc/libbitcoin_server_a-blockchain.o
  CXX      rpc/libbitcoin_server_a-masternode.o
  CXX      rpc/libbitcoin_server_a-governance.o
  CXX      rpc/libbitcoin_server_a-mining.o
  CXX      rpc/libbitcoin_server_a-misc.o
  CXX      rpc/libbitcoin_server_a-net.o
  CXX      rpc/libbitcoin_server_a-rawtransaction.o
  CXX      rpc/libbitcoin_server_a-server.o
  CXX      script/libbitcoin_server_a-sigcache.o
  CXX      libbitcoin_server_a-sendalert.o
  CXX      libbitcoin_server_a-timedata.o
  CXX      libbitcoin_server_a-torcontrol.o
  CXX      libbitcoin_server_a-txdb.o
  CXX      libbitcoin_server_a-txmempool.o
In file included from /usr/include/boost/assert.hpp:58:0,
                 from /usr/include/boost/signals2/detail/lwm_pthreads.hpp:21,
                 from /usr/include/boost/signals2/mutex.hpp:30,
                 from /usr/include/boost/signals2/signal.hpp:31,
                 from util.h:30,
                 from txmempool.cpp:16:
txmempool.cpp: In member function ‘void CTxMemPool::check(const CCoinsViewCache*) const’:
txmempool.cpp:816:49: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             assert(it->GetSizeWithDescendants() >= childSizes + it->GetTxSize());
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      libbitcoin_server_a-validation.o
  CXX      libbitcoin_server_a-validationinterface.o
  CXX      libbitcoin_server_a-versionbits.o
  AR       libbitcoin_server.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      libbitcoin_common_a-amount.o
  CXX      libbitcoin_common_a-arith_uint256.o
  CXX      libbitcoin_common_a-base58.o
  CXX      libbitcoin_common_a-bip39.o
  CXX      libbitcoin_common_a-chainparams.o
  CXX      libbitcoin_common_a-coins.o
  CXX      libbitcoin_common_a-compressor.o
  CXX      consensus/libbitcoin_common_a-merkle.o
  CXX      libbitcoin_common_a-core_read.o
  CXX      libbitcoin_common_a-core_write.o
  CXX      libbitcoin_common_a-hash.o
  CXX      libbitcoin_common_a-hdchain.o
  CXX      libbitcoin_common_a-key.o
  CXX      libbitcoin_common_a-keystore.o
  CXX      libbitcoin_common_a-netaddress.o
  CXX      libbitcoin_common_a-netbase.o
  CXX      primitives/libbitcoin_common_a-block.o
  CXX      primitives/libbitcoin_common_a-transaction.o
  CXX      libbitcoin_common_a-protocol.o
  CXX      libbitcoin_common_a-pubkey.o
  CXX      libbitcoin_common_a-scheduler.o
  CXX      script/libbitcoin_common_a-interpreter.o
  CXX      script/libbitcoin_common_a-script.o
  CXX      script/libbitcoin_common_a-script_error.o
  CXX      script/libbitcoin_common_a-sign.o
  CXX      script/libbitcoin_common_a-standard.o
  CXX      libbitcoin_common_a-spork.o
  AR       libbitcoin_common.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
make[3]: Entering directory '/home/src/compiles/dash/src/univalue'
  CXX      lib/libunivalue_la-univalue.lo
  CXX      lib/libunivalue_la-univalue_read.lo
  CXX      lib/libunivalue_la-univalue_write.lo
  CXXLD    libunivalue.la
ar: `u' modifier ignored since `D' is the default (see `U')
make[3]: Leaving directory '/home/src/compiles/dash/src/univalue'
  CXX      support/libbitcoin_util_a-pagelocker.o
  CXX      libbitcoin_util_a-chainparamsbase.o
  CXX      libbitcoin_util_a-clientversion.o
  CXX      compat/libbitcoin_util_a-glibc_sanity.o
  CXX      compat/libbitcoin_util_a-glibcxx_sanity.o
  CXX      compat/libbitcoin_util_a-strnlen.o
  CXX      libbitcoin_util_a-random.o
  CXX      rpc/libbitcoin_util_a-protocol.o
  CXX      support/libbitcoin_util_a-cleanse.o
  CXX      libbitcoin_util_a-sync.o
  CXX      libbitcoin_util_a-uint256.o
  CXX      libbitcoin_util_a-threadinterrupt.o
  CXX      libbitcoin_util_a-util.o
  CXX      libbitcoin_util_a-utilmoneystr.o
  CXX      libbitcoin_util_a-utilstrencodings.o
  CXX      libbitcoin_util_a-utiltime.o
  AR       libbitcoin_util.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      crypto/crypto_libbitcoin_crypto_a-hmac_sha256.o
  CXX      crypto/crypto_libbitcoin_crypto_a-hmac_sha512.o
  CXX      crypto/crypto_libbitcoin_crypto_a-ripemd160.o
  CC       crypto/crypto_libbitcoin_crypto_a-aes_helper.o
  CXX      crypto/crypto_libbitcoin_crypto_a-sha1.o
  CXX      crypto/crypto_libbitcoin_crypto_a-sha256.o
  CXX      crypto/crypto_libbitcoin_crypto_a-sha512.o
  CC       crypto/crypto_libbitcoin_crypto_a-blake.o
  CC       crypto/crypto_libbitcoin_crypto_a-bmw.o
  CC       crypto/crypto_libbitcoin_crypto_a-cubehash.o
  CC       crypto/crypto_libbitcoin_crypto_a-echo.o
  CC       crypto/crypto_libbitcoin_crypto_a-groestl.o
  CC       crypto/crypto_libbitcoin_crypto_a-jh.o
  CC       crypto/crypto_libbitcoin_crypto_a-keccak.o
  CC       crypto/crypto_libbitcoin_crypto_a-luffa.o
  CC       crypto/crypto_libbitcoin_crypto_a-shavite.o
  CC       crypto/crypto_libbitcoin_crypto_a-simd.o
  CC       crypto/crypto_libbitcoin_crypto_a-skein.o
  AR       crypto/libbitcoin_crypto.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      leveldb/db/leveldb_libleveldb_a-builder.o
  CXX      leveldb/db/leveldb_libleveldb_a-c.o
  CXX      leveldb/db/leveldb_libleveldb_a-dbformat.o
  CXX      leveldb/db/leveldb_libleveldb_a-db_impl.o
  CXX      leveldb/db/leveldb_libleveldb_a-db_iter.o
  CXX      leveldb/db/leveldb_libleveldb_a-dumpfile.o
  CXX      leveldb/db/leveldb_libleveldb_a-filename.o
  CXX      leveldb/db/leveldb_libleveldb_a-log_reader.o
  CXX      leveldb/db/leveldb_libleveldb_a-log_writer.o
  CXX      leveldb/db/leveldb_libleveldb_a-memtable.o
In file included from ./leveldb/util/arena.h:9:0,
                 from ./leveldb/db/skiplist.h:33,
                 from ./leveldb/db/memtable.h:11,
                 from leveldb/db/memtable.cc:5:
leveldb/db/memtable.cc: In member function ‘void leveldb::MemTable::Add(leveldb::SequenceNumber, leveldb::ValueType, const leveldb::Slice&, const leveldb::Slice&)’:
leveldb/db/memtable.cc:104:31: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
   assert((p + val_size) - buf == encoded_len);
          ~~~~~~~~~~~~~~~~~~~~~^~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      leveldb/db/leveldb_libleveldb_a-repair.o
  CXX      leveldb/db/leveldb_libleveldb_a-table_cache.o
  CXX      leveldb/db/leveldb_libleveldb_a-version_edit.o
  CXX      leveldb/db/leveldb_libleveldb_a-version_set.o
  CXX      leveldb/db/leveldb_libleveldb_a-write_batch.o
  CXX      leveldb/table/leveldb_libleveldb_a-block_builder.o
  CXX      leveldb/table/leveldb_libleveldb_a-block.o
  CXX      leveldb/table/leveldb_libleveldb_a-filter_block.o
leveldb/table/filter_block.cc: In member function ‘bool leveldb::FilterBlockReader::KeyMayMatch(uint64_t, const leveldb::Slice&)’:
leveldb/table/filter_block.cc:100:33: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (start <= limit && limit <= (offset_ - data_)) {
                           ~~~~~~^~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      leveldb/table/leveldb_libleveldb_a-format.o
  CXX      leveldb/table/leveldb_libleveldb_a-iterator.o
  CXX      leveldb/table/leveldb_libleveldb_a-merger.o
  CXX      leveldb/table/leveldb_libleveldb_a-table_builder.o
  CXX      leveldb/table/leveldb_libleveldb_a-table.o
  CXX      leveldb/table/leveldb_libleveldb_a-two_level_iterator.o
  CXX      leveldb/util/leveldb_libleveldb_a-arena.o
  CXX      leveldb/util/leveldb_libleveldb_a-bloom.o
leveldb/util/bloom.cc: In member function ‘virtual void leveldb::{anonymous}::BloomFilterPolicy::CreateFilter(const leveldb::Slice*, int, std::__cxx11::string*) const’:
leveldb/util/bloom.cc:50:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (size_t i = 0; i < n; i++) {
                        ~~^~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      leveldb/util/leveldb_libleveldb_a-cache.o
  CXX      leveldb/util/leveldb_libleveldb_a-coding.o
  CXX      leveldb/util/leveldb_libleveldb_a-comparator.o
  CXX      leveldb/util/leveldb_libleveldb_a-crc32c.o
  CXX      leveldb/util/leveldb_libleveldb_a-env.o
  CXX      leveldb/util/leveldb_libleveldb_a-env_posix.o
  CXX      leveldb/util/leveldb_libleveldb_a-env_win.o
  CXX      leveldb/util/leveldb_libleveldb_a-filter_policy.o
  CXX      leveldb/util/leveldb_libleveldb_a-hash.o
  CXX      leveldb/util/leveldb_libleveldb_a-histogram.o
  CXX      leveldb/util/leveldb_libleveldb_a-logging.o
leveldb/util/logging.cc: In function ‘bool leveldb::ConsumeDecimalNumber(leveldb::Slice*, uint64_t*)’:
leveldb/util/logging.cc:58:40: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
           (v == kMaxUint64/10 && delta > kMaxUint64%10)) {
                                  ~~~~~~^~~~~~~~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      leveldb/util/leveldb_libleveldb_a-options.o
  CXX      leveldb/util/leveldb_libleveldb_a-status.o
  CXX      leveldb/port/leveldb_libleveldb_a-port_posix.o
  AR       leveldb/libleveldb.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      leveldb/helpers/memenv/leveldb_libmemenv_a-memenv.o
  AR       leveldb/libmemenv.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      libbitcoin_wallet_a-activemasternode.o
  CXX      libbitcoin_wallet_a-dsnotificationinterface.o
  CXX      libbitcoin_wallet_a-instantx.o
  CXX      libbitcoin_wallet_a-masternode.o
  CXX      libbitcoin_wallet_a-masternode-payments.o
  CXX      libbitcoin_wallet_a-masternode-sync.o
  CXX      libbitcoin_wallet_a-masternodeconfig.o
  CXX      libbitcoin_wallet_a-masternodeman.o
masternodeman.cpp: In member function ‘bool CMasternodeMan::GetMasternodeByRank(int, masternode_info_t&, int, int)’:
masternodeman.cpp:713:36: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (vecMasternodeScores.size() < nRankIn)
         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      libbitcoin_wallet_a-keepass.o
  CXX      libbitcoin_wallet_a-privatesend-client.o
  CXX      libbitcoin_wallet_a-privatesend-util.o
  CXX      wallet/libbitcoin_wallet_a-crypter.o
  CXX      wallet/libbitcoin_wallet_a-db.o
  CXX      wallet/libbitcoin_wallet_a-rpcdump.o
  CXX      wallet/libbitcoin_wallet_a-rpcwallet.o
  CXX      wallet/libbitcoin_wallet_a-wallet.o
wallet/wallet.cpp: In member function ‘bool CWallet::AddToWallet(const CWalletTx&, bool, CWalletDB*)’:
wallet/wallet.cpp:959:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for(int i = 0; i < wtx.vout.size(); ++i) {
                            ~~^~~~~~~~~~~~~~~~~
wallet/wallet.cpp: In member function ‘DBErrors CWallet::LoadWallet(bool&)’:
wallet/wallet.cpp:3639:30: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
             for(int i = 0; i < pair.second.vout.size(); ++i) {
                            ~~^~~~~~~~~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      wallet/libbitcoin_wallet_a-wallet_ismine.o
  CXX      wallet/libbitcoin_wallet_a-walletdb.o
  CXX      policy/libbitcoin_wallet_a-rbf.o
  AR       libbitcoin_wallet.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXXLD    dashd
  CXX      dash_cli-dash-cli.o
  CXX      rpc/libbitcoin_cli_a-client.o
  AR       libbitcoin_cli.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXXLD    dash-cli
  CXX      dash_tx-dash-tx.o
  CXXLD    dash-tx
Generated test/data/script_tests.json.h
Generated test/data/base58_keys_valid.json.h
Generated test/data/base58_encode_decode.json.h
Generated test/data/base58_keys_invalid.json.h
Generated test/data/bip39_vectors.json.h
Generated test/data/tx_invalid.json.h
Generated test/data/tx_valid.json.h
Generated test/data/sighash.json.h
Generated test/data/alertTests.raw.h
  CXX      test/test_test_dash-arith_uint256_tests.o
  CXX      test/test_test_dash-addrman_tests.o
  CXX      test/test_test_dash-alert_tests.o
  CXX      test/test_test_dash-allocator_tests.o
  CXX      test/test_test_dash-base32_tests.o
  CXX      test/test_test_dash-base58_tests.o
  CXX      test/test_test_dash-base64_tests.o
  CXX      test/test_test_dash-bip32_tests.o
  CXX      test/test_test_dash-bip39_tests.o
  CXX      test/test_test_dash-bloom_tests.o
  CXX      test/test_test_dash-bswap_tests.o
  CXX      test/test_test_dash-cachemap_tests.o
  CXX      test/test_test_dash-cachemultimap_tests.o
  CXX      test/test_test_dash-checkblock_tests.o
  CXX      test/test_test_dash-coins_tests.o
  CXX      test/test_test_dash-compress_tests.o
  CXX      test/test_test_dash-crypto_tests.o
  CXX      test/test_test_dash-DoS_tests.o
  CXX      test/test_test_dash-getarg_tests.o
  CXX      test/test_test_dash-governance_validators_tests.o
  CXX      test/test_test_dash-hash_tests.o
  CXX      test/test_test_dash-key_tests.o
  CXX      test/test_test_dash-limitedmap_tests.o
  CXX      test/test_test_dash-dbwrapper_tests.o
  CXX      test/test_test_dash-main_tests.o
test/main_tests.cpp:35:13: warning: ‘void main_tests::TestBlockSubsidyHalvings(int)’ defined but not used [-Wunused-function]
 static void TestBlockSubsidyHalvings(int nSubsidyHalvingInterval)
             ^~~~~~~~~~~~~~~~~~~~~~~~
test/main_tests.cpp:17:13: warning: ‘void main_tests::TestBlockSubsidyHalvings(const Consensus::Params&)’ defined but not used [-Wunused-function]
 static void TestBlockSubsidyHalvings(const Consensus::Params& consensusParams)
             ^~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      test/test_test_dash-mempool_tests.o
  CXX      test/test_test_dash-merkle_tests.o
  CXX      test/test_test_dash-miner_tests.o
  CXX      test/test_test_dash-multisig_tests.o
  CXX      test/test_test_dash-net_tests.o
  CXX      test/test_test_dash-netbase_tests.o
  CXX      test/test_test_dash-pmt_tests.o
  CXX      test/test_test_dash-policyestimator_tests.o
  CXX      test/test_test_dash-pow_tests.o
  CXX      test/test_test_dash-prevector_tests.o
  CXX      test/test_test_dash-ratecheck_tests.o
  CXX      test/test_test_dash-reverselock_tests.o
  CXX      test/test_test_dash-rpc_tests.o
  CXX      test/test_test_dash-sanity_tests.o
  CXX      test/test_test_dash-scheduler_tests.o
  CXX      test/test_test_dash-script_P2SH_tests.o
  CXX      test/test_test_dash-script_P2PKH_tests.o
In file included from /usr/include/boost/test/test_tools.hpp:45:0,
                 from /usr/include/boost/test/unit_test.hpp:18,
                 from test/script_P2PKH_tests.cpp:8:
test/script_P2PKH_tests.cpp: In member function ‘void script_P2PKH_tests::IsPayToPublicKeyHash::test_method()’:
test/script_P2PKH_tests.cpp:50:18: warning: array subscript is above array bounds [-Warray-bounds]
     BOOST_CHECK(!CScript(missing2, missing2+sizeof(missing)).IsPayToPublicKeyHash());
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test/script_P2PKH_tests.cpp:50:18: warning: array subscript is above array bounds [-Warray-bounds]
test/script_P2PKH_tests.cpp:55:18: warning: array subscript is above array bounds [-Warray-bounds]
     BOOST_CHECK(!CScript(tooshort, tooshort+sizeof(direct)).IsPayToPublicKeyHash());
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
test/script_P2PKH_tests.cpp:55:18: warning: array subscript is above array bounds [-Warray-bounds]
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  CXX      test/test_test_dash-script_tests.o
  CXX      test/test_test_dash-scriptnum_tests.o
  CXX      test/test_test_dash-serialize_tests.o
  CXX      test/test_test_dash-sighash_tests.o
  CXX      test/test_test_dash-sigopcount_tests.o
  CXX      test/test_test_dash-skiplist_tests.o
  CXX      test/test_test_dash-streams_tests.o
  CXX      test/test_test_dash-test_dash.o
  CXX      test/test_test_dash-timedata_tests.o
  CXX      test/test_test_dash-transaction_tests.o
  CXX      test/test_test_dash-txvalidationcache_tests.o
  CXX      test/test_test_dash-versionbits_tests.o
  CXX      test/test_test_dash-uint256_tests.o
  CXX      test/test_test_dash-univalue_tests.o
  CXX      test/test_test_dash-util_tests.o
  CXX      test/test_test_dash-accounting_tests.o
  CXX      wallet/test/test_test_dash-wallet_tests.o
  CXX      test/test_test_dash-rpc_wallet_tests.o
  CXXLD    test/test_dash
  CXX      bench/bench_bench_dash-bench_dash.o
  CXX      bench/bench_bench_dash-bench.o
  CXX      bench/bench_bench_dash-Examples.o
  CXXLD    bench/bench_dash
  GEN      qt/forms/ui_addressbookpage.h
  GEN      qt/forms/ui_askpassphrasedialog.h
  GEN      qt/forms/ui_coincontroldialog.h
  GEN      qt/forms/ui_darksendconfig.h
  GEN      qt/forms/ui_editaddressdialog.h
  GEN      qt/forms/ui_helpmessagedialog.h
  GEN      qt/forms/ui_intro.h
  GEN      qt/forms/ui_modaloverlay.h
  GEN      qt/forms/ui_masternodelist.h
  GEN      qt/forms/ui_openuridialog.h
  GEN      qt/forms/ui_optionsdialog.h
  GEN      qt/forms/ui_overviewpage.h
  GEN      qt/forms/ui_receivecoinsdialog.h
  GEN      qt/forms/ui_receiverequestdialog.h
  GEN      qt/forms/ui_debugwindow.h
  GEN      qt/forms/ui_sendcoinsdialog.h
  GEN      qt/forms/ui_sendcoinsentry.h
  GEN      qt/forms/ui_signverifymessagedialog.h
  GEN      qt/forms/ui_transactiondescdialog.h
  GEN      qt/paymentrequest.pb.h
[libprotobuf WARNING google/protobuf/compiler/parser.cc:547] No syntax specified for the proto file: paymentrequest.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
  GEN      qt/dash.moc
  GEN      qt/bitcoinamountfield.moc
  GEN      qt/intro.moc
  GEN      qt/overviewpage.moc
  GEN      qt/rpcconsole.moc
  CXX      qt/qt_dash_qt-dash.o
  CXX      qt/qt_libbitcoinqt_a-bantablemodel.o
  CXX      qt/qt_libbitcoinqt_a-bitcoinaddressvalidator.o
  CXX      qt/qt_libbitcoinqt_a-bitcoinamountfield.o
  CXX      qt/qt_libbitcoinqt_a-bitcoingui.o
  CXX      qt/qt_libbitcoinqt_a-bitcoinunits.o
  CXX      qt/qt_libbitcoinqt_a-clientmodel.o
  CXX      qt/qt_libbitcoinqt_a-csvmodelwriter.o
  CXX      qt/qt_libbitcoinqt_a-guiutil.o
  CXX      qt/qt_libbitcoinqt_a-intro.o
  CXX      qt/qt_libbitcoinqt_a-modaloverlay.o
  CXX      qt/qt_libbitcoinqt_a-networkstyle.o
  CXX      qt/qt_libbitcoinqt_a-notificator.o
  CXX      qt/qt_libbitcoinqt_a-optionsdialog.o
  CXX      qt/qt_libbitcoinqt_a-optionsmodel.o
  CXX      qt/qt_libbitcoinqt_a-peertablemodel.o
  CXX      qt/qt_libbitcoinqt_a-platformstyle.o
  CXX      qt/qt_libbitcoinqt_a-qvalidatedlineedit.o
  CXX      qt/qt_libbitcoinqt_a-qvaluecombobox.o
  CXX      qt/qt_libbitcoinqt_a-rpcconsole.o
  CXX      qt/qt_libbitcoinqt_a-splashscreen.o
  CXX      qt/qt_libbitcoinqt_a-trafficgraphdata.o
  CXX      qt/qt_libbitcoinqt_a-trafficgraphwidget.o
  CXX      qt/qt_libbitcoinqt_a-utilitydialog.o
  CXX      qt/qt_libbitcoinqt_a-addressbookpage.o
  CXX      qt/qt_libbitcoinqt_a-addresstablemodel.o
  CXX      qt/qt_libbitcoinqt_a-askpassphrasedialog.o
  CXX      qt/qt_libbitcoinqt_a-coincontroldialog.o
  CXX      qt/qt_libbitcoinqt_a-coincontroltreewidget.o
  CXX      qt/qt_libbitcoinqt_a-darksendconfig.o
  CXX      qt/qt_libbitcoinqt_a-editaddressdialog.o
  CXX      qt/qt_libbitcoinqt_a-masternodelist.o
  CXX      qt/qt_libbitcoinqt_a-openuridialog.o
  CXX      qt/qt_libbitcoinqt_a-overviewpage.o
  CXX      qt/qt_libbitcoinqt_a-paymentrequestplus.o
  CXX      qt/qt_libbitcoinqt_a-paymentserver.o
  CXX      qt/qt_libbitcoinqt_a-receivecoinsdialog.o
  CXX      qt/qt_libbitcoinqt_a-receiverequestdialog.o
  CXX      qt/qt_libbitcoinqt_a-recentrequeststablemodel.o
  CXX      qt/qt_libbitcoinqt_a-sendcoinsdialog.o
  CXX      qt/qt_libbitcoinqt_a-sendcoinsentry.o
  CXX      qt/qt_libbitcoinqt_a-signverifymessagedialog.o
  CXX      qt/qt_libbitcoinqt_a-transactiondesc.o
  CXX      qt/qt_libbitcoinqt_a-transactiondescdialog.o
  CXX      qt/qt_libbitcoinqt_a-transactionfilterproxy.o
  CXX      qt/qt_libbitcoinqt_a-transactionrecord.o
  CXX      qt/qt_libbitcoinqt_a-transactiontablemodel.o
  CXX      qt/qt_libbitcoinqt_a-transactionview.o
  CXX      qt/qt_libbitcoinqt_a-walletframe.o
  CXX      qt/qt_libbitcoinqt_a-walletmodel.o
  CXX      qt/qt_libbitcoinqt_a-walletmodeltransaction.o
  CXX      qt/qt_libbitcoinqt_a-walletview.o
  GEN      qt/moc_addressbookpage.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_addressbookpage.o
  GEN      qt/moc_addresstablemodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_addresstablemodel.o
  GEN      qt/moc_askpassphrasedialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_askpassphrasedialog.o
  GEN      qt/moc_bantablemodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_bantablemodel.o
  GEN      qt/moc_bitcoinaddressvalidator.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_bitcoinaddressvalidator.o
  GEN      qt/moc_bitcoinamountfield.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_bitcoinamountfield.o
  GEN      qt/moc_bitcoingui.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_bitcoingui.o
  GEN      qt/moc_bitcoinunits.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_bitcoinunits.o
  GEN      qt/moc_clientmodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_clientmodel.o
  GEN      qt/moc_coincontroldialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_coincontroldialog.o
  GEN      qt/moc_coincontroltreewidget.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_coincontroltreewidget.o
  GEN      qt/moc_csvmodelwriter.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_csvmodelwriter.o
  GEN      qt/moc_darksendconfig.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_darksendconfig.o
  GEN      qt/moc_editaddressdialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_editaddressdialog.o
  GEN      qt/moc_guiutil.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_guiutil.o
  GEN      qt/moc_intro.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_intro.o
  GEN      qt/moc_macdockiconhandler.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_macdockiconhandler.o
  GEN      qt/moc_macnotificationhandler.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_macnotificationhandler.o
  GEN      qt/moc_modaloverlay.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_modaloverlay.o
  GEN      qt/moc_masternodelist.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_masternodelist.o
  GEN      qt/moc_notificator.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_notificator.o
  GEN      qt/moc_openuridialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_openuridialog.o
  GEN      qt/moc_optionsdialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_optionsdialog.o
  GEN      qt/moc_optionsmodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_optionsmodel.o
  GEN      qt/moc_overviewpage.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_overviewpage.o
  GEN      qt/moc_peertablemodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_peertablemodel.o
  GEN      qt/moc_paymentserver.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_paymentserver.o
  GEN      qt/moc_qvalidatedlineedit.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_qvalidatedlineedit.o
  GEN      qt/moc_qvaluecombobox.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_qvaluecombobox.o
  GEN      qt/moc_receivecoinsdialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_receivecoinsdialog.o
  GEN      qt/moc_receiverequestdialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_receiverequestdialog.o
  GEN      qt/moc_recentrequeststablemodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_recentrequeststablemodel.o
  GEN      qt/moc_rpcconsole.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_rpcconsole.o
  GEN      qt/moc_sendcoinsdialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_sendcoinsdialog.o
  GEN      qt/moc_sendcoinsentry.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_sendcoinsentry.o
  GEN      qt/moc_signverifymessagedialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_signverifymessagedialog.o
  GEN      qt/moc_splashscreen.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_splashscreen.o
  GEN      qt/moc_trafficgraphwidget.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_trafficgraphwidget.o
  GEN      qt/moc_transactiondesc.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_transactiondesc.o
  GEN      qt/moc_transactiondescdialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_transactiondescdialog.o
  GEN      qt/moc_transactionfilterproxy.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_transactionfilterproxy.o
  GEN      qt/moc_transactiontablemodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_transactiontablemodel.o
  GEN      qt/moc_transactionview.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_transactionview.o
  GEN      qt/moc_utilitydialog.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_utilitydialog.o
  GEN      qt/moc_walletframe.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_walletframe.o
  GEN      qt/moc_walletmodel.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_walletmodel.o
  GEN      qt/moc_walletview.cpp
  CXX      qt/qt_libbitcoinqt_a-moc_walletview.o
  CXX      qt/qt_libbitcoinqt_a-paymentrequest.pb.o
  GEN      qt/qrc_dash.cpp
  CXX      qt/qt_libbitcoinqt_a-qrc_dash.o
  GEN      qt/locale/dash_bg.qm
  GEN      qt/locale/dash_de.qm
  GEN      qt/locale/dash_en.qm
  GEN      qt/locale/dash_es.qm
  GEN      qt/locale/dash_fi.qm
  GEN      qt/locale/dash_fr.qm
  GEN      qt/locale/dash_it.qm
  GEN      qt/locale/dash_ja.qm
  GEN      qt/locale/dash_pl.qm
  GEN      qt/locale/dash_pt.qm
  GEN      qt/locale/dash_ru.qm
Removed plural forms as the target language has less forms.
If this sounds wrong, possibly the target language is not set or recognized.
  GEN      qt/locale/dash_sk.qm
  GEN      qt/locale/dash_sv.qm
  GEN      qt/locale/dash_vi.qm
  GEN      qt/locale/dash_zh_CN.qm
  GEN      qt/locale/dash_zh_TW.qm
  GEN      qt/qrc_dash_locale.cpp
  CXX      qt/qt_libbitcoinqt_a-qrc_dash_locale.o
  AR       qt/libbitcoinqt.a
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  OBJCXXLD qt/dash-qt
  CXX      qt/test/qt_test_test_dash_qt-compattests.o
  CXX      qt/test/qt_test_test_dash_qt-test_main.o
  CXX      qt/test/qt_test_test_dash_qt-uritests.o
  CXX      qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o
  CXX      qt/test/qt_test_test_dash_qt-paymentservertests.o
qt/test/paymentservertests.cpp: In member function ‘void PaymentServerTests::paymentServerTests()’:
qt/test/paymentservertests.cpp:65:6: warning: stack protector not protecting local variables: variable length buffer [-Wstack-protector]
 void PaymentServerTests::paymentServerTests()
      ^~~~~~~~~~~~~~~~~~
At global scope:
cc1plus: warning: unrecognized command line option ‘-Wno-implicit-fallthrough’
cc1plus: warning: unrecognized command line option ‘-Wno-self-assign’
  GEN      qt/test/moc_compattests.cpp
  CXX      qt/test/qt_test_test_dash_qt-moc_compattests.o
  GEN      qt/test/moc_trafficgraphdatatests.cpp
  CXX      qt/test/qt_test_test_dash_qt-moc_trafficgraphdatatests.o
  GEN      qt/test/moc_uritests.cpp
  CXX      qt/test/qt_test_test_dash_qt-moc_uritests.o
  GEN      qt/test/moc_paymentservertests.cpp
  CXX      qt/test/qt_test_test_dash_qt-moc_paymentservertests.o
  CXXLD    qt/test/test_dash-qt
qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o: In function `TrafficGraphDataTests::averageBandwidthEvery2EmptyTest()':
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:221: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:220: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o: In function `TrafficGraphDataTests::averageBandwidthTest()':
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:190: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:189: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:198: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o:/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:197: more undefined references to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)' follow
collect2: error: ld returned 1 exit status
Makefile:3723: recipe for target 'qt/test/test_dash-qt' failed
make[2]: *** [qt/test/test_dash-qt] Error 1
make[2]: Leaving directory '/home/src/compiles/dash/src'
Makefile:9432: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/src/compiles/dash/src'
Makefile:656: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
src@d:~/compiles/dash$ 

```

I repeat here the important part of the log file:

  CXXLD    qt/test/test_dash-qt
qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o: In function `TrafficGraphDataTests::averageBandwidthEvery2EmptyTest()':
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:221: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:220: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o: In function `TrafficGraphDataTests::averageBandwidthTest()':
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:190: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:189: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:198: undefined reference to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)'
qt/test/qt_test_test_dash_qt-trafficgraphdatatests.o:/home/src/compiles/dash/src/qt/test/trafficgraphdatatests.cpp:197: more undefined references to `bool QTest::qCompare<float, double>(float const&, double const&, char const*, char const*, char const*, int)' follow
collect2: error: ld returned 1 exit status

--- a/src/qt/test/trafficgraphdatatests.cpp
+++ b/src/qt/test/trafficgraphdatatests.cpp
@@ -186,16 +186,16 @@ void TrafficGraphDataTests::averageBandwidthTest()
     }
     QCOMPARE(trafficGraphData.getCurrentRangeQueueWithAverageBandwidth().size(), TrafficGraphData::DESIRED_DATA_SAMPLES);
     for(auto& sample : trafficGraphData.getCurrentRangeQueueWithAverageBandwidth()){
-        QCOMPARE(sample.in, 1.0);
-        QCOMPARE(sample.out, 1.0);
+        QCOMPARE(sample.in, 1.0f);
+        QCOMPARE(sample.out, 1.0f);
     }
 
     trafficGraphData.switchRange(TrafficGraphData::Range_10m);
 
     QCOMPARE(trafficGraphData.getCurrentRangeQueueWithAverageBandwidth().size(), TrafficGraphData::DESIRED_DATA_SAMPLES / 2);
     for(auto& sample : trafficGraphData.getCurrentRangeQueueWithAverageBandwidth()){
-        QCOMPARE(sample.in, 1.0);
-        QCOMPARE(sample.out, 1.0);
+        QCOMPARE(sample.in, 1.0f);
+        QCOMPARE(sample.out, 1.0f);
     }
 }
 
@@ -217,7 +217,7 @@ void TrafficGraphDataTests::averageBandwidthEvery2EmptyTest()
 
     QCOMPARE(trafficGraphData.getCurrentRangeQueueWithAverageBandwidth().size(), TrafficGraphData::DESIRED_DATA_SAMPLES / 2);
     for(auto& sample : trafficGraphData.getCurrentRangeQueueWithAverageBandwidth()){
-        QCOMPARE(sample.in, 0.5);
-        QCOMPARE(sample.out, 0.5);
+        QCOMPARE(sample.in, 0.5f);
+        QCOMPARE(sample.out, 0.5f);
     }
 }

